src/field/arena.ts

```diff
@@ -810,8 +810,15 @@ export class Arena {
       existingTag.onOverlap();
 
       if (ENTRY_HAZARD_ARENA_TAG_TYPES.includes(existingTag.tagType)) {
-        const { tagType, side, turnCount, layers, maxLayers } = existingTag as EntryHazardTag;
-        this.eventTarget.dispatchEvent(new TagAddedEvent(tagType, side, turnCount, layers, maxLayers));
+        const { layers, maxLayers } = existingTag as EntryHazardTag;
+        const tagEvent = new TagAddedEvent(
+          existingTag.tagType,
+          existingTag.side,
+          existingTag.turnCount,
+          layers,
+          maxLayers,
```

Comment:
ngl it feels weird to deconstruct half of it, I'd prefer to keep things consistent because at a glance no one would have any clue why it is done that way
```suggestion
        const tagEvent = new TagAddedEvent(
          existingTag.tagType,
          existingTag.side,
          existingTag.turnCount,
          existingTag.layers,
          existingTag.maxLayers,
```

Reply:
I made this suggestion on another file in #1322, maybe something similar can be done here?
```ts
for (const tag of globalScene.arena.tags) {
  const { tagType, side, turnCount } = tag;
  const { layers, maxLayers } = tag instanceof EntryHazardTag ? tag : {};
  globalScene.arena.eventTarget.dispatchEvent(
    new TagAddedEvent(tagType, side, turnCount, layers ?? 0, maxLayers ?? 0);
  ); 
}
```

Reply:
> ngl it feels weird to deconstruct half of it, I'd prefer to keep things consistent because at a glance no one would have any clue why it is done that way

If it's not done that way then you have to do `(existingTag as EntryHazardTag).layers, (existingTag as EntryHazardTag).maxLayers`. I can convert it to use inner's suggestion though.

Reply:
Actually in the first place I changed this due to `noShadow` so I can't do that.

Reply:
Okay I renamed the method params instead of removing the destructuring, and simplified the second one as well.
Also fixed the TSDocs while I was at it.